### PR TITLE
Add js heap size limit & tests

### PR DESCRIPTION
**Pull request by letmaik**
Part of https://github.com/microsoft/CCF/issues/1459. Next step will be to make the hard-coded limits configurable.

### CODE REVIEW

**Review by achamayou — APPROVED** [review]

**Review by eddyashton — APPROVED** [review]

**Comment by letmaik** [comment]
The SGX stack overflow test still crashes, likely because the enclave runs out of stack space. It has 4 MiB in total, and quickjs's limit is set to 1 MiB. That means for it to crash the rest of CCF needs to use more than 3 MiB? I wonder if there is a way to check stack usage at the point when the endpoint handler is called, simply to fail hard if the requested quickjs stack limit + existing usage would exceed the enclave limit.

**Comment by letmaik** [comment]
Turns out the stack check wasn't enabled for the enclave build. @achamayou I had to patch quickjs.c to enable this (see changes). As part of https://github.com/microsoft/CCF/pull/2162 I moved the patches to a separate folder, and although this patch doesn't allow this easily I think the next-best thing would be to let the build fail if `CONFIG_STACK_CHECK` is not defined (which would indicated a faulty merge of newer quickjs versions).

File: cmake/quickjs.cmake [hunk 1]
```diff
@@ -29,7 +29,7 @@ if("sgx" IN_LIST COMPILE_TARGETS)
   )
   target_compile_options(
     quickjs.enclave PUBLIC -nostdinc -DCONFIG_VERSION="${QUICKJS_VERSION}"
-                           -DEMSCRIPTEN
+                           -DEMSCRIPTEN -DENABLE_STACK_CHECK
```
Comment (achamayou) [inline cmake/quickjs.cmake:32]: define CONFIG_STACK_CHECK and avoid having to patch quickjs?

Comment (letmaik) [reply cmake/quickjs.cmake]: Genius.

File: src/apps/js_generic/js_generic.cpp [hunk 2]
```diff
@@ -969,6 +969,7 @@ namespace ccfapp
       JSAutoFreeRuntime auto_free_rt(rt);
 
       JS_SetMaxStackSize(rt, 1024 * 1024);
+      JS_SetMemoryLimit(rt, 20 * 1024 * 1024);
```
Comment (achamayou) [inline src/apps/js_generic/js_generic.cpp:972]: This seems low, we typically run with 500Mb of heap, can we make it 100Mb at least?

**Review by achamayou — APPROVED** [review]

**Pull request merged**